src/genie/libs/parser/asa/show_route.py

```diff
@@ -326,3 +326,156 @@ def cli(self, output=None):
+class ShowRouteBgp(ShowRouteSchema):
+    """Parser for
+        * show route bgp
+    """
+
+    cli_command = 'show route bgp'
+
+    def cli(self, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+
+        ret_dict = {}
+        # strip this patter from the original text
+        # Codes: L - Local, C - connected, S - static, I - IGRP, R - RIP, M - mobile, B - BGP
+        # D - EIGRP, E - EGP, EX - EIGRP external, O - OSPF, I - IGRP, IA - OSPF inter area
+        # N1 - OSPF NSSA external type 1, N2 - OSPF NSSA external type 2
+        # E1 - OSPF external type 1, E2 - OSPF external type 2, E - EGP
+        # i - IS-IS, L1 - IS-IS level-1, L2 - IS-IS level-2, ia - IS-IS inter area
+        # * - candidate default, su - IS-IS summary, U - per-user static route, o - ODR
+        # P - periodic downloaded static route, + - replicated route
+        # SI - Static InterVRF
+        p1 = re.sub(r'(?ms)(Codes:.+?)replicated\sroute(\s*SI\s-\sStatic\sInterVRF)?', '', out)
+
+        lines = [x.strip() for x in p1.splitlines()]
+        entries = dict()
+        last_entry = str()
+        clean_lines = list()
+
+        for line in lines:
+            if re.match(r'(^[A-Z]{1,2})', line):
+                entries[line] = list()
+                last_entry = line
+            else:
+                if last_entry == '':
+                    continue
+
+                entries[last_entry].append(line)
+
+        for k, v in entries.items():
+            clean_lines.append(' '.join(k.split()) + " " + " ".join(v))
+
+        # clean_lines now holds a list of entries in a single line (easier for parsing)
+        # Gateway of last resort is X.Y.Z.44 to network 0.0.0.0
+        # B        X.Y.Z.137 255.255.255.255 [20/0] via X.Y.Z.15, 1w0d
+        # B        X.Y.Z.138 255.255.255.255 [20/0] via X.Y.Z.15, 1w0d
+        # B        X.Y.Z.140 255.255.255.255 [20/0] via X.Y.Z.45, 1w0d
+        # B        X.Y.Z.141 255.255.255.255 [20/0] via X.Y.Z.45, 1w0d
+        # B        X.Y.Z.160 255.255.255.255 [20/5] via X.Y.Z.21, 1w0d
+        # B        X.Y.Z.161 255.255.255.255 [20/5] via X.Y.Z.21, 1w0d
+        # B        X.Y.Z.164 255.255.255.255 [20/5] via X.Y.Z.21, 1w0d
+        # B        X.Y.Z.165 255.255.255.255 [20/5] via X.Y.Z.21, 1w0d
+        # B        X.Y.Z.166 255.255.255.255 [20/0] via X.Y.Z.33, 2d17h
+        # B        X.Y.Z.252 255.255.255.255 [20/0] via X.Y.Z.17, 1w0d
+        # B        X.Y.Z.253 255.255.255.255 [20/0] via X.Y.Z.17, 1w0d
+
+        # [110/11] via 10.20.192.3, 1w1d, wan3 [110/11] via 10.20.192.4, 1w1d, wan4
+
+        # B X.Y.Z.253 255.255.255.255 [20/0] via X.Y.X.17, 1w0d
+        p2 = re.compile(r'(?P<code>\S+)\s(?P<network>\S+)\s(?P<subnet>\S+)\s\[(?P<route_preference>[\d\/]+)\]')
+        # [20/0] via 172.25.141.2, 7w0d
+        p3 = re.compile(r'\[(?P<route_preference>[\d\/]+)\]\svia\s+(?P<next_hop>\S+),\s(?P<date>\S+)')
```

Comment:
The regular expressions you have defined here kind of make the entire last 50-ish lines obsolete. They'll never match the other lines, so cleaning them out seems like a waste of effort :/